gunpowder/nodes/snapshot.py

```diff
@@ -90,6 +90,10 @@ def __init__(
 
     def setup(self):
 
+        for key, _ in self.additional_request.items():
+            assert key in self.dataset_names, (
```

Comment:
This should be how it works, only requesting what you want to save, but there are some nodes that do not yet properly handle their dependencies. For example on train nodes, you need to request both the inputs and the outputs. You can't just request the outputs yet since depending on your model, you need to request input with some context. In this case you might want to add both input and output to your additional_request, but only save the output.

Reply:
Do you mean that the other way around or maybe I am misunderstanding something. The train node consumes (and maybe passes on) the input and creates the output. So it should request the inputs and not the outputs, the request is passed upstream, isn't it?

Reply:
So the upstream node that produces A erroneously does not request B? I see your point, and I understand that the dependency mechanism is still a work in progress, but relying on some downstream node that might not even be used to fix that bug sounds a bit unreliable.

Reply:
Good point